outlines/structure.md

```diff
@@ -0,0 +1,56 @@
+# Application structure and code style
+
+1. What's different between Meteor and other kinds of apps?
+    1. All JavaScript, enables code sharing
+    2. Structure is flexible
+    3. Currently everything is about magic file names
+    4. App scope vs. package scope vs. file scope variables
+2. Client vs. server vs. both
+3. Smooth path from prototype to production, three sizes of apps, S, M, L
+4. General stuff
+    1. Filenames - named the same as the export
+        1. Every file exports one thing
+    2. Avoid depending on load order via namespacing and Meteor.startup
+        1. If you depend on load order, your best bet is a package
+    3. Naming guidelines for everything
+        1. https://github.com/meteor/guide/issues/9#issuecomment-148800085
+    4. Templates should have a CSS class that matches the template name, and other Blaze suggestions
+    5. Meteor ES2015 style guide (link to it, don't reproduce all of it)
+    6. Perhaps this should include copypasta boilerplate - a standard template/component, a standard publication, standard route, etc? Basically the stuff a scaffolding framework would generate for you?
+5. Small app
+    1. Directory structure around features, not client/server
```

Comment:
So I have also a [model package](https://github.com/peerlibrary/meteor-peerdb) and I tried few times to combine all these into one class (I like OOP programming if you haven't noticed). So what I was thinking in the past was something like:

```
class Post extends Document

class Post.List extends BlazeComponent
  @register 'Post.List'

  onCreated: ->
    @subscribe 'Post.list'

Post.new = (doc) ->
  Meteor.call 'Post.new', doc

Post._new = (doc) ->
  @insert doc

FlowRouter.route '/posts',
  name: 'Post.list'
  action: (params, queryParams) ->
    BlazeLayout.render 'Layout',
      main: 'Post.List'

Meteor.methods 'Post.new', Post._new.bind(Post)

Meteor.publish 'Post.list' ->
  Post.documents.find()
```

(I use `Post` instead of `Posts` because they are documents, not collections. And I use `Post.List` instead of `Post.list` because it is a component, not template. And because `Post.list` might be a method.)

But then, there are always some exceptions. Methods which have to operate on both Posts and Comments and the same time. Publish functions which return multiple collections (like all Posts and Comments tagged with the same tag). Where to put then those?

Maybe we should say that publish functions should always return only one collection? But then that limits a lot thins, no?

BTW, I am not proposing that this is structure and naming which should be proposed in the guide. Things change a bit once you have models and components. I think `Posts` is better when you have only collections, no models. And `Posts.list` is better when you have templates and no components. (In this way when you do `{{> Posts.list}}` you know that you are including a template, and `{{> Posts.List}}` is a component.)